mod-source-record-manager-server/src/main/java/org/folio/services/EventDrivenChunkProcessingServiceImpl.java

```diff
@@ -154,4 +161,31 @@ private boolean isRecordReadyToSend(Record record) {
       });
     return promise.future();
   }
+
+  private Future<Boolean> updateJobExecutionIfAllSourceChunksMarkedAsError(String jobExecutionId, OkapiConnectionParams params) {
+    return jobExecutionSourceChunkDao.get("jobExecutionId=" + jobExecutionId + " AND last=true", 0, 1, params.getTenantId())
+      .compose(chunks -> isNotEmpty(chunks) ? jobExecutionSourceChunkDao.isAllChunksProcessed(jobExecutionId, params.getTenantId()) : Future.succeededFuture(false))
+      .compose(isAllChunksError -> {
+        if (isAllChunksError) {
+          StatusDto statusDto = new StatusDto().withStatus(StatusDto.Status.ERROR).withErrorStatus(StatusDto.ErrorStatus.RECORD_UPDATE_ERROR);
+          return jobExecutionProgressService.getByJobExecutionId(jobExecutionId, params.getTenantId())
+            .compose(progress -> updateJobExecutionState(jobExecutionId, progress, statusDto, params));
+        }
+        return Future.succeededFuture(false);
+      });
+  }
+
+  private Future<Boolean> updateJobExecutionState(String jobExecutionId, JobExecutionProgress progress, StatusDto statusDto, OkapiConnectionParams params) {
```

Comment:
don't we have such method in JobExecutionService or smth?

Reply:
no, we have not